frontend/src/admin/components/dropdown-menu/dropdown-menu.tsx

```diff
@@ -0,0 +1,37 @@
+import React, {useState} from "react";
+import DropdownMenuIcon from "../dropdown-menu-icon/dropdown-menu-icon";
+import "./dropdown-menu.scss";
+
+interface DropdownMenuProps {
+    children: React.ReactNode
+}
+
+export default function DropdownMenu( {children}: DropdownMenuProps) {
+    const [showDropdown, setShowDropdown] = useState(false);
+
+    function toggleDropdown() {
+        setShowDropdown(!showDropdown);
+    }
+
+    function hideDropdown(event: React.FocusEvent<HTMLButtonElement>): void {
+        if (event.currentTarget === event.target) {
```

Comment:
In what case does this become false?
I.e what is the difference between `.currentTarget` and `.target`?

Reply:
The condition event.currentTarget === event.target is checking whether the event was triggered by the button itself or one of its children. If the event was triggered by the button itself, the condition will be true, and the code inside the if block will be executed. Otherwise, if the event was triggered by one of the button's children, the condition will be false, and the code inside the if block will be skipped.

The currentTarget property refers to the element on which the event listener is attached.

 The target property of the event object refers to the actual element that triggered the event.

Reply:
A suggestion could be to extract the condition out into a function and give it a proper name of what it checks for.